langchain/src/schema/runnable/passthrough.ts

```diff
@@ -1,6 +1,85 @@
-import { Runnable } from "./base.js";
+import { IterableReadableStream } from "../../util/stream.js";
+import { Runnable, RunnableMap } from "./base.js";
 import type { RunnableConfig } from "./config.js";
 
+/**
+ * A runnable that assigns key-value pairs to inputs of type `Record<string, unknown>`.
+ */
+export class RunnableAssign<
+  RunInput = unknown,
+  RunOutput = unknown,
+  CallOptions extends RunnableConfig = RunnableConfig
+> extends Runnable<RunInput, RunOutput> {
+  lc_namespace = ["langchain", "schema", "runnable"];
+
+  mapper: RunnableMap<RunInput>;
+
+  constructor(mapper: RunnableMap<RunInput>) {
+    super();
+    this.mapper = mapper;
+  }
+
+  async invoke(input: RunInput, options?: Partial<CallOptions>) {
+    const mapperResult = await this.mapper.invoke(input, options);
+
+    return {
+      ...input,
+      ...mapperResult,
+    } as RunOutput;
```

Comment:
Cast is necessary? Can we just add it as the return type above?

Reply:
We have to cast since the result of .invoke is `Record<string, any>` and not RunOutput

The runnable.invoke types are `Runnable<RunInput, any>` so it's being typed as `Record<string, any>` on the RunnableMap side

Reply:
Got it, we should try to fix the typing at some point